tests/test_synced_collection.py

```diff
@@ -0,0 +1,587 @@
+        synced_list.reset([1, 2, 1])
+        synced_list.remove(1)
+        assert len(synced_list) == 2
+        assert synced_list[0] == 2
+        assert synced_list[1] == 1
+
+    def test_call(self, synced_list):
+        synced_list.reset([1, 2])
+        assert len(synced_list) == 2
+        assert isinstance(synced_list(), list)
+        assert synced_list() == [1, 2]
+
+    def test_update_recursive(self, synced_list):
+        synced_list.reset([{'a': 1}, 'b', [1, 2, 3]])
+        assert synced_list == [{'a': 1}, 'b', [1, 2, 3]]
+        data = ['a', 'b', [1, 2, 4], 'd']
+        self.store(data)
+        assert synced_list == data
+        data1 = ['a', 'b']
+        self.store(data1)
+        assert synced_list == data1
+
+        # inavlid data in file
+        data2 = {'a': 1}
+        self.store(data2)
+        with pytest.raises(ValueError):
+            synced_list.load()
+
+    def test_reopen(self, synced_list, testdata):
+        synced_list.append(testdata)
+        synced_list.sync()
+        del synced_list  # possibly unsafe
+        synced_list2 = self._cls(**self._backend_kwargs)
+        synced_list2.load()
+        assert len(synced_list2) == 1
+        assert synced_list2[0] == testdata
+
+    def test_copy_as_list(self, synced_list, testdata):
+        synced_list.append(testdata)
+        assert synced_list[0] == testdata
+        copy = list(synced_list)
+        del synced_list
+        assert copy[0] == testdata
+
+    def test_repr(self, synced_list):
+        repr(synced_list)
+        p = eval(repr(synced_list))
+        assert repr(p) == repr(synced_list)
+        assert p == synced_list
+
+    def test_str(self, synced_list):
+        str(synced_list) == str(synced_list.to_base())
+
+    def test_nested_list(self, synced_list):
+        synced_list.reset([1, 2, 3])
+        synced_list.append([2, 4])
+        child1 = synced_list[3]
+        child2 = synced_list[3]
+        assert child1 == child2
+        assert isinstance(child1, type(child2))
+        assert isinstance(child1, self._cls)
+        assert id(child1) == id(child2)
+        child1.append(1)
+        assert child2[2] == child1[2]
+        assert child1 == child2
+        assert len(synced_list) == 4
+        assert isinstance(child1, type(child2))
+        assert isinstance(child1, self._cls)
+        assert id(child1) == id(child2)
+        del child1[0]
+        assert child1 == child2
+        assert len(synced_list) == 4
+        assert isinstance(child1, type(child2))
+        assert isinstance(child1, self._cls)
+        assert id(child1) == id(child2)
+
+    def test_nested_list_with_dict(self, synced_list):
+        synced_list.reset([{'a': [1, 2, 3, 4]}])
+        child1 = synced_list[0]
+        child2 = synced_list[0]['a']
+        assert isinstance(child2, SyncedCollection)
+        assert isinstance(child1, SyncedCollection)
+
+
+class TestJSONListWriteConcern(TestJSONList):
+
+    _write_concern = True
```

Comment:
I think it would be preferable for clarity to override the fixture also for the control of this fixture constructor parameter.